src/controllers/error.controller.ts

```diff
@@ -17,7 +17,8 @@ const notFoundHandler = (req: Request, res: Response, next: NextFunction) => {
  * last handler in the chain for it to work.
  */
 const errorHandler = async (err: unknown, req: Request, res: Response, next: NextFunction) => {
-  logger.error(errorMessages.ERROR_500, err);
+  logger.error(errorMessages.ERROR_500);
```

Comment:
in our overseas-entities-web error controller we also log the stack trace, wondering if that might be a useful upgrade to add (and if errorRequest function is available on that version of logger perhaps use that and pass the request in as well)

`logger.errorRequest(req, `An error has occurred. Re-routing to the error screen - ${err.stack}`);`

see https://github.com/companieshouse/overseas-entities-web/blob/main/src/controllers/error.controller.ts#L19C1-L20C1

Reply:
Tried to update this to use an Error object and print out the stack but it comes through as undefined. Presumably this ties in with updating the service to actually use a ch-structured-logger?

[2024-02-29T14:43:08.830] [INFO] extensions-web - Error caught posting a chosen reason
[2024-02-29T14:43:08.840] [ERROR] extensions-web - An error has occurred. Re-routing to the error screen: undefined

Reply:
don't think it's logger related, should be a field on the error object.  But if it isn't working then we'll stick with what we have.